code/controllers/subsystem/ping.dm

```diff
@@ -0,0 +1,32 @@
+SUBSYSTEM_DEF(ping)
+	name = "Ping"
+	priority = FIRE_PRIORITY_PING
+	wait = 15 SECONDS
+	flags = SS_NO_INIT
+	runlevels = RUNLEVEL_LOBBY|RUNLEVEL_SETUP|RUNLEVEL_GAME|RUNLEVEL_POSTGAME
+
+	var/list/currentrun = list()
+
+/datum/controller/subsystem/ping/stat_entry()
+	..("P:[length(GLOB.clients)]")
+
+
+/datum/controller/subsystem/ping/fire(resumed = FALSE)
+	if(!resumed)
+		src.currentrun = GLOB.clients.Copy()
+
+	//cache for sanic speed (lists are references anyways)
+	var/list/currentrun = src.currentrun
+
+	while(length(currentrun))
+		var/client/C = currentrun[length(currentrun)]
+		currentrun.len--
+
+		if(!C?.chatOutput?.loaded)
+			if(MC_TICK_CHECK)
+				return
```

Comment:
Wouldnt it be better to have a global for loaded chats instead of checking whether each client has a loaded chat?

Reply:
I am not sure, this is how it did it before.

Reply:
on second thought this might be better

Reply:
It just works.